main/exercise/ptest_exercise_report.php

```diff
@@ -0,0 +1,504 @@
+$action_links = '';
+// Generating group list
+$group_list = GroupManager::get_group_list();
+$group_parameters = [
+    'group_all:'.get_lang('All'),
+    'group_none:'.get_lang('None'),
+];
+
+foreach ($group_list as $group) {
+    $group_parameters[] = $group['id'].':'.$group['name'];
+}
+if (!empty($group_parameters)) {
+    $group_parameters = implode(';', $group_parameters);
+}
+
+$officialCodeInList = api_get_setting('show_official_code_exercise_result_list');
+
+if ($is_allowedToEdit || $is_tutor) {
+    // The order is important you need to check the the $column variable in the model.ajax.php file
+    $columns = [
+        get_lang('FirstName'),
+        get_lang('LastName'),
+        get_lang('LoginName'),
+        get_lang('Group'),
+        get_lang('Duration').' ('.get_lang('MinMinute').')',
+        get_lang('StartDate'),
+        get_lang('EndDate'),
+        get_lang('IP'),
+        get_lang('ToolLearnpath'),
+        get_lang('Actions'),
+    ];
+
+    if ($officialCodeInList === 'true') {
+        $columns = array_merge([get_lang('OfficialCode')], $columns);
+    }
+
+    // Column config
+    $column_model = [
+        ['name' => 'firstname', 'index' => 'firstname', 'width' => '50', 'align' => 'left', 'search' => 'true'],
+        ['name' => 'lastname', 'index' => 'lastname', 'width' => '50', 'align' => 'left', 'formatter' => 'action_formatter', 'search' => 'true'],
+        [
+            'name' => 'login',
+            'index' => 'username',
+            'width' => '40',
+            'align' => 'left',
+            'search' => 'true',
+            'hidden' => api_get_configuration_value('exercise_attempts_report_show_username') ? 'false' : 'true',
+        ],
+        [
+            'name' => 'group_name',
+            'index' => 'group_id',
+            'width' => '40',
+            'align' => 'left',
+            'search' => 'true',
+            'stype' => 'select',
+            //for the bottom bar
+            'searchoptions' => [
+                'defaultValue' => 'group_all',
+                'value' => $group_parameters,
+            ],
+            //for the top bar
+            'editoptions' => ['value' => $group_parameters],
+        ],
+        ['name' => 'duration', 'index' => 'exe_duration', 'width' => '30', 'align' => 'left', 'search' => 'true'],
+        ['name' => 'start_date', 'index' => 'start_date', 'width' => '60', 'align' => 'left', 'search' => 'true'],
+        ['name' => 'exe_date', 'index' => 'exe_date', 'width' => '60', 'align' => 'left', 'search' => 'true'],
+        ['name' => 'ip', 'index' => 'user_ip', 'width' => '40', 'align' => 'center', 'search' => 'true'],
+        ['name' => 'lp', 'index' => 'orig_lp_id', 'width' => '60', 'align' => 'left', 'search' => 'false'],
+        ['name' => 'actions', 'index' => 'actions', 'width' => '60', 'align' => 'left', 'search' => 'false', 'sortable' => 'false'],
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 132 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)